### PR TITLE
Added Jira extra functionality

### PR DESCRIPTION
**Pull request by Dinnerspy**
added new parameters
jira.child (Boolean)
jira.ParentUrl(String)
cx-flow.ApplicationRepoOnly (Boolean)
checkmarx.nonamespace: (Boolean)

Jira Key = ApplicationRepoOnly
Project team's in Jira sometimes have to have more than one repo to support a single application. We need to ensure that cxFlow supports a Jira key that includes: Application and Repo name.

multilayer Jira support
Avoid duplicate issues from being created across the program Better metrics for teams vs ARTs
Ability to use Jira at multiple levels to manage "Application Security Bugs"

Naming Convention issue
A bug that caused the webhook to create new projects in checkmarks as it could not read the cmd line runners projects.

### CODE REVIEW

**Comment by Dinnerspy** [comment]
testing is being finished up right now

**Comment by Dinnerspy** [comment]
Testing was a came back positive.

File: src/main/java/com/checkmarx/flow/service/CxService.java [hunk 1]
```diff
@@ -1168,7 +1168,11 @@ public String createTeam(String parentTeamId, String teamName) throws MachinaExc
         try {
             session = cxLegacyService.login(cxProperties.getUsername(), cxProperties.getPassword());
             cxLegacyService.createTeam(session, parentTeamId, teamName);
+            if(!cxProperties.getNonamespace() ){
```
[the remark this answers is not in the document]
Comment (Dinnerspy) [reply src/main/java/com/checkmarx/flow/service/CxService.java]: As we want the cmd line approach and the GIT lab webhook to update the same project in CheckMarx

We use the “override team name” functionality from the command line and it works great. when we do the same thing from the webhook it takes it but it always appends the gitlab team name at the end. we don’t want it to do this.  we want it to update the same project if we override please take the full override.

[the remark this answers is not in the document]
Comment (kmcdon83) [reply src/main/java/com/checkmarx/flow/service/CxService.java]: I understand what you are trying to accomplish, the logic should not be here though.  This is a simple function to create the team and return its new id.  What you are describing is more "business" logic.

I suggest adding to the FlowService, at the point this function is called to create the team check if your property is set, and if so use the parent id that is already known (or call getTeamId providing the exact same logic as you have here)

[the remark this answers is not in the document]
Comment (kmcdon83) [reply src/main/java/com/checkmarx/flow/service/CxService.java]: Are you certain that setting the cx.multi-tenant: false is not what you want?  This indicates you do not want to create a team based on the namespace.  If you review the logic, if this is set, the team specified in the property or override is used, and nothing is created.

checkmarx:
    multi-tenant: false

[the remark this answers is not in the document]
Comment (Dinnerspy) [reply src/main/java/com/checkmarx/flow/service/CxService.java]: ok we can do that then

**Comment by kmcdon83** [comment]
Review completed.

**Comment by kmcdon83** [comment]
@Dinnerspy going forward please work from the following repository: https://github.com/checkmarx-ts/cx-flow.git and the develop branch.  This is going to be our field development repository going forward.  I have applied these changes to this repo now and will incorporate them into the next release, which should happen shortly.

**Pull request merged**